rfc/system/5103-offloaded-literal.md

```diff
@@ -0,0 +1,69 @@
+# [RFC] Offloaded Raw Literals
+
+**Authors:**
+
+- @wild-endeavor
+- @EngHabu
+
+## 1 Executive Summary
+
+Flyte depends on a series of `inputs.pb` and `outputs.pb` files to do communication between nodes. This has typically served us well, except for the occasional map task that produces a large Literal output. We sometimes also run into this issue for large dataclasses. This RFC proposes a mechanism that allows the offloading of any Literal, which would be done only of course for now for size reasons.
+
+## 2 Motivation
+A [cursory search](https://discuss.flyte.org/?threads%5Bquery%5D=LIMIT_EXCEEDED) of Slack history shows a few times that this has come up before (and I remember some other instances, I think that search term just wasn't included). This is something that we've historically addressed by just increasing the size of the grpc message that's allowed, but this is an unsustainable solution.
+
+## 3 Proposed Implementation
+
+### 3.1 Offloaded Literal IDL
+To the `Literal` [message](https://github.com/flyteorg/flyte/blob/cb6384ac6ea60f8b9421a71cfda4279f3579d3cb/flyteidl/protos/flyteidl/core/literals.proto#L95), add a new field called `starp` that will point to a location in the "metadata" bucket of the Flyte backend. The offloaded bytes should be deserialzable into a `Literal` object.
+
+Questions: How will things like metadata be handled? Should they be merged? What should be in the `value` field of the main parent Literal?
+
+### 3.2 Flyte Propeller
+* When writing map task outputs, depending on the size, Propeller will need to offload the LiteralCollection after constructing it, and create a new Literal for downstream tasks to use, with the 
+* Also Propeller will need to check the flytekit version of the map task. If it's an older version (i.e. before the change proposed in this RFC), and it's large enough to need to be offloaded, it should fail the task. The assumption here is that if the map task is of the older version then downstream tasks will probably also be of those older versions which won't know how to resolved these offloaded literals. 
+
+### 3.3 Flytekit & Copilot
+Flytekit and Copilot will both need to detect that a Literal has been offloaded and know to download it.
+
+For large outputs (like large maps of large dataclasses), Flytekit should also know how to offload the data. This should be done transparently to the user. How will propeller know to fail though if propeller hasn't been updated?
+
+### 3.4 Other Implications
+Does console need to change at all?
```

Comment:
As a user clicking on the `inputs` tab of a node, I'd expect to see the string representation of the Literal including the blob storage uri the literal has been offloaded to.

It would be nice to see the type of the literal as is for example already the case when e.g. a pytorch module or pickled object is offloaded (with the only difference that the offloading happens in the respective type transformer in flytekit and not in propeller of course).